text/master-based-development.md

```diff
@@ -0,0 +1,29 @@
+# Master Based Development (MBD)
```

Comment:
I like this being in the initial RFC. I might add some issues for similar topics that should be added.